contracts/payloads/IGP111/PayloadIGP111.sol

```diff
@@ -0,0 +1,396 @@
+            VAULT_FACTORY.setVaultAuth(
+                syrupUSDT__GHO_VAULT,
+                TEAM_MULTISIG,
+                false
+            );
+        }
+    }
+
+    /// @notice Action 2: Revenue collection for buyback
+    function action2() internal isActionSkippable(2) {
+        {
+            address[] memory tokens = new address[](14);
+            tokens[0] = USDT_ADDRESS;
+            tokens[1] = wstETH_ADDRESS;
+            tokens[2] = ETH_ADDRESS;
+            tokens[3] = USDC_ADDRESS;
+            tokens[4] = sUSDe_ADDRESS;
+            tokens[5] = cbBTC_ADDRESS;
+            tokens[6] = WBTC_ADDRESS;
+            tokens[7] = GHO_ADDRESS;
+            tokens[8] = USDe_ADDRESS;
+            tokens[9] = wstUSR_ADDRESS;
+            tokens[10] = ezETH_ADDRESS;
+            tokens[11] = lBTC_ADDRESS;
+            tokens[12] = USDTb_ADDRESS;
+            tokens[13] = RLP_ADDRESS;
+
+            LIQUIDITY.collectRevenue(tokens);
+        }
+        {
+            address[] memory tokens = new address[](14);
+            uint256[] memory amounts = new uint256[](14);
+
+            tokens[0] = USDT_ADDRESS;
+            amounts[0] = IERC20(USDT_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 10;
+
+            tokens[1] = wstETH_ADDRESS;
+            amounts[1] = IERC20(wstETH_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 0.1 ether;
+
+            tokens[2] = ETH_ADDRESS;
+            amounts[2] = address(FLUID_RESERVE).balance - 0.1 ether; // 0.1 ETH
+
+            tokens[3] = USDC_ADDRESS;
+            amounts[3] = IERC20(USDC_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 10;
+
+            tokens[4] = sUSDe_ADDRESS;
+            amounts[4] = IERC20(sUSDe_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 0.1 ether;
+
+            tokens[5] = cbBTC_ADDRESS;
+            amounts[5] = IERC20(cbBTC_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 10;
+
+            tokens[6] = WBTC_ADDRESS;
+            amounts[6] = IERC20(WBTC_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 10;
+
+            tokens[7] = GHO_ADDRESS;
+            amounts[7] = IERC20(GHO_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 10;
+
+            tokens[8] = USDe_ADDRESS;
+            amounts[8] = IERC20(USDe_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 10;
+
+            tokens[9] = wstUSR_ADDRESS;
+            amounts[9] = IERC20(wstUSR_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 10;
+
+            tokens[10] = ezETH_ADDRESS;
+            amounts[10] = IERC20(ezETH_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 0.1 ether;
+
+            tokens[11] = lBTC_ADDRESS;
+            amounts[11] = IERC20(lBTC_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 10;
+
+            tokens[12] = USDTb_ADDRESS;
+            amounts[12] = IERC20(USDTb_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 10;
+
+            tokens[13] = RLP_ADDRESS;
+            amounts[13] = IERC20(RLP_ADDRESS).balanceOf(address(FLUID_RESERVE)) - 10;
+
+            FLUID_RESERVE.withdrawFunds(tokens, amounts, TEAM_MULTISIG);
```

Comment:
if we switch up action3 and action2, then only need to trigger collect revenue and we can remove this whole part of withdrawFunds logic